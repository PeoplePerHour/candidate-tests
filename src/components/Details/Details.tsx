import * as React from "react";
import * as models from "../../models";

interface Iprops {
  character: models.ICharacter;
}

const Details = (props: Iprops) => {
  const { character } = props;
  return (
    <div>
      <div className="d-flex fd-col fa-center card details">
        <div>
          <h1>{character.name}</h1>
        </div>
        <img src={character.image} alt="" />

        <h3>Status: {character.status}</h3>
        <h3>Species: {character.species}</h3>
        <h3>gender: {character.gender}</h3>
        <h3>type: {character.type}</h3>
        <h3>Url: {character.url}</h3>
        <h3>Episodes</h3>
        {character.episode ? (
          <ul>
            {" "}
            {character.episode.map((e, i) => {
              return <li key={i}>{e}</li>;
            })}
          </ul>
        ) : null}
      </div>
    </div>
  );
};

export default Details;
