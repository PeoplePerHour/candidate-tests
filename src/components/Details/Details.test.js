
import * as  React from 'react';
import * as ReactDOM from 'react-dom';

import Details from './Details'

const data = [
    {
      id: 1,
      name: 'Rick Sanchez',
      status: 'Alive',
      species: 'Human',
      type: '',
      gender: 'Male',
      origin: {
        name: 'Earth (C-137)',
        url: 'https://rickandmortyapi.com/api/location/1'
      },
      location: {
        name: 'Earth (Replacement Dimension)',
        url: 'https://rickandmortyapi.com/api/location/20'
      },
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      episode: [
        'https://rickandmortyapi.com/api/episode/1',
        'https://rickandmortyapi.com/api/episode/2',
      ],
      url: 'https://rickandmortyapi.com/api/character/1',
      created: '2017-11-04T18:48:46.250Z'
    },
]

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Details character={data}/>, div);
    expect(div.firstChild).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
})
