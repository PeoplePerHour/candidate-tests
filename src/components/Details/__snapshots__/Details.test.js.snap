// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="d-flex fd-col fa-center card details"
  >
    <div>
      <h1 />
    </div>
    <img
      alt=""
    />
    <h3>
      Status: 
    </h3>
    <h3>
      Species: 
    </h3>
    <h3>
      gender: 
    </h3>
    <h3>
      type: 
    </h3>
    <h3>
      Url: 
    </h3>
    <h3>
      Episodes
    </h3>
  </div>
</div>
`;
